services/web/server/src/simcore_service_webserver/projects/projects_db_utils.py

```diff
@@ -46,51 +48,73 @@ def check_project_permissions(
     project: Union[ProjectProxy, ProjectDict],
     user_id: int,
     user_groups: list[RowProxy],
-    permission: Permission,
+    permission: str,
 ) -> None:
+    """
+    :raises ProjectInvalidRightsError if check fails
+    """
+
     if not permission:
         return
 
-    needed_permissions = permission.split("|")
+    requested_actions = set(permission.split("|"))
+    assert set(requested_actions).issubset(set(PermissionStr.__args__))  # nosec
 
-    # compute access rights by order of priority all group > organizations > primary
-    primary_group = next(
-        filter(lambda x: x.get("type") == GroupType.PRIMARY, user_groups), None
-    )
-    standard_groups = filter(lambda x: x.get("type") == GroupType.STANDARD, user_groups)
+    #
+    # Get primary_gid, standard_gids and everyone_gid for user_id
+    #
     all_group = next(
         filter(lambda x: x.get("type") == GroupType.EVERYONE, user_groups), None
     )
-    if primary_group is None or all_group is None:
-        # the user groups is missing entries
+    if all_group is None:
         raise ProjectInvalidRightsError(user_id, project.get("uuid"))
 
+    everyone_gid = str(all_group["gid"])
+
+    if user_id == ANY_USER_ID_SENTINEL:
+        primary_gid = None
+        standard_gids = []
+
+    else:
+        primary_group = next(
+            filter(lambda x: x.get("type") == GroupType.PRIMARY, user_groups), None
+        )
+        if primary_group is None:
+            # the user groups is missing entries
+            raise ProjectInvalidRightsError(user_id, project.get("uuid"))
+
+        standard_groups = filter(
+            lambda x: x.get("type") == GroupType.STANDARD, user_groups
+        )
+
+        primary_gid = str(primary_group["gid"])
+        standard_gids = [str(group["gid"]) for group in standard_groups]
+
+    #
+    # Composes access rights by order of priority all group > organizations > primary
+    #
     project_access_rights = deepcopy(project.get("access_rights", {}))
 
-    # compute access rights
-    no_access_rights = {"read": False, "write": False, "delete": False}
-    computed_permissions = project_access_rights.get(
-        str(all_group["gid"]), no_access_rights
+    # access rights for everyone
+    can = project_access_rights.get(
```

Comment:
TIP: maybe consider a more eloquent name here like: `everyone_can`

Reply:
this is not only for "everyone" but "can" is composed with other group access rights. 
It could be "user_can" ...